internal/executor/executor.go

```diff
@@ -119,13 +119,14 @@ func (executor *Executor) RunBuild(ctx context.Context) {
 		return
 	}
 
-	scriptEnvironment := getScriptEnvironment(executor, response.Environment)
+	executor.env.MergeLazy(getScriptEnvironment(executor, response.Environment), false,
```

Comment:
Naming of `noLazy` argument puzzled me more than bringing clarity. I still not quite understand what the naming means? I think I understand what you wanted to achinve but I can't come up with better naming.

Alternatively, have you tried to something like this:

```diff
--- a/internal/executor/executor.go
+++ b/internal/executor/executor.go
@@ -125,7 +125,8 @@ func (executor *Executor) RunBuild(ctx context.Context) {
 
        var vaultUnboxer *vaultunboxer.VaultUnboxer
 
-       for key, value := range scriptEnvironment {
+       expandedScriptEnvironment := environment.ExpandEnvironmentRecursively(scriptEnvironment)
+       for key, value := range expandedScriptEnvironment {
                boxedValue, err := vaultunboxer.NewBoxedValue(value)
                if err != nil {
                        if errors.Is(err, vaultunboxer.ErrNotABoxedValue) {
@@ -142,7 +143,7 @@ func (executor *Executor) RunBuild(ctx context.Context) {
                if vaultUnboxer == nil {
                        log.Println("Found at least one VAULT[...] environment variable, initializing Vault client")
 
-                       vaultUnboxer, err = vaultunboxer.NewFromEnvironment(ctx, environment.New(scriptEnvironment))
+                       vaultUnboxer, err = vaultunboxer.NewFromEnvironment(ctx, environment.New(expandedScriptEnvironment))
                        if err != nil {
                                message := fmt.Sprintf("failed to initialize a Vault client: %v", err)
                                log.Println(message)
```

This logic seems will respect expanded values for the Vault unboxing and then pass them to merging into the `env`.

Reply:
Good point, see https://github.com/cirruslabs/cirrus-ci-agent/pull/333/commits/df837bcb536ed23f0c18c6ab87f4ac8da50b8dd8.